File: 040_Read/button1.cpp
#include <iostream>   // Include pour affichage et clavier
#include <signal.h>   // signal de fin de programme
#include <stdlib.h>   // librairie standard c++
#include <unistd.h>   // usleep
#include <wiringPi.h> // Include WiringPi library!

using namespace std;

// Pin number declarations. We're using the Broadcom chip pin numbers.
const int gpio23 = 23; // Regular LED - Broadcom pin 23, P1 pin 16
const int gpio20 = 20; // Regular LED - Broadcom pin 20, P1 pin 38

// Fonction de fin déclenchée par CTRL-C
void fin(int sig)
{
	// Désactive les résistances de pull up, pull down
	pullUpDnControl(gpio23, PUD_OFF);
	pullUpDnControl(gpio20, PUD_OFF);
	cout << "FIN signal: " << sig << endl;
	exit(0);
}

int main(void)
{
	// Ecoute du CTRL-C avec fonction à lancer
	signal(SIGINT, fin);
	
	// Setup 
	wiringPiSetupGpio(); // Initialize wiringPi -- using Broadcom pin numbers

	pinMode(gpio23, INPUT);
	pinMode(gpio20, INPUT);
	pullUpDnControl(gpio23, PUD_DOWN);
	pullUpDnControl(gpio20, PUD_UP);

	cout << "Press CTRL+C pour sortir" << endl;

	// Loop (while(1)):
	int compteur = 0;
	while(1)
	{
		cout << compteur << " GPIO23 = " << digitalRead(23) << " - GPIO20 = " << digitalRead(20) << endl;
		usleep(500000); // Wait 500ms
		compteur++;
	}
}

File: 040_Read/button1.py
import RPi.GPIO as GPIO    # utilisation GPIO
from time import sleep     # utilisation sleep  
GPIO.setmode(GPIO.BCM)     # numérotation BCM
GPIO.setup(23, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)  # GPIO23 en entrée avec résistance pull down  
GPIO.setup(20, GPIO.IN, pull_up_down=GPIO.PUD_UP)    # GPIO20 en entrée avec résistance pull up  
  
try:  
    compteur = 0
    while True:            # Jusqu'à CTRL+C  
        print compteur, " GPIO23 = ", GPIO.input(23), " - GPIO20 = ", GPIO.input(20)
        sleep(0.5)         # Attend 0.5s
        compteur += 1;
  
finally:                   # Bloc de fin aprés CTRL-C 
    GPIO.cleanup()         # Nettoyage des GPIO (force les GPIO utilisés en entrée sans résistance)
    print "FIN"